components/raftstore/src/store/peer.rs

```diff
@@ -2618,6 +2624,53 @@ impl Peer {
             }
         }
     }
+
+    pub fn bcast_check_stale_peer_message<T: Transport, C>(&mut self, ctx: &mut PollContext<T, C>) {
+        ctx.need_flush_trans = true;
+        if self.check_stale_conf_ver < self.region().get_region_epoch().get_conf_ver() {
+            self.check_stale_conf_ver = self.region().get_region_epoch().get_conf_ver();
+            self.check_stale_peers = self.region().get_peers().to_vec();
+        }
+        for peer in &self.check_stale_peers {
+            if peer.get_id() == self.peer_id() {
+                continue;
+            }
+            let mut send_msg = RaftMessage::default();
+            send_msg.set_region_id(self.region_id);
+            send_msg.set_from_peer(self.peer.clone());
+            send_msg.set_region_epoch(self.region().get_region_epoch().clone());
+            send_msg.set_to_peer(peer.clone());
+            let extra_msg = send_msg.mut_extra_msg();
+            extra_msg.set_type(ExtraMessageType::MsgCheckStalePeer);
+            if let Err(e) = ctx.trans.send(send_msg) {
+                error!(
+                    "failed to send check stale peer message";
+                    "region_id" => self.region_id,
+                    "peer_id" => self.peer.get_id(),
+                    "target_peer_id" => peer.get_id(),
+                    "target_store_id" => peer.get_store_id(),
+                    "err" => ?e,
+                );
+            }
+        }
+    }
+
+    pub fn on_check_stale_peer_response<T: Transport, C>(
+        &mut self,
+        ctx: &mut PollContext<T, C>,
+        check_conf_ver: u64,
+        check_peers: &[metapb::Peer],
+    ) {
+        if self.check_stale_conf_ver < self.region().get_region_epoch().get_conf_ver() {
```

Comment:
This seems unnecessary.

Reply:
Because the check-stale-peer msg should use the latest peer list.